README.md

```diff
@@ -375,6 +375,32 @@ SearchResponse(ok,developer,1853997,1,10,1,185400,newest,List(Content(politics/b
 scala> client.shutdown()
 ```
 
+## Serialisation / Deserialisation
+
+To serialise the models returned from the client, you can use the JSON encoder in [content-api-models](https://github.com/guardian/content-api-models). You will need to import circe as a dependency in `build.sbt` –
+
+```
+val circeVersion = "0.12.3"
+
+libraryDependencies ++= Seq(
+  "io.circe" %% "circe-core",
+  "io.circe" %% "circe-generic",
+  "io.circe" %% "circe-parser"
+).map(_ % circeVersion)
+```
+
+... and then call the relevant method on the model.
+
+```scala
+import com.gu.contentapi.json.CirceEncoders._
+import io.circe.syntax._
+
+// ... later ...
+
+val query = ContentApiClient.search.q("An example query")
+client.getResponse(query).asJson.toString
```

Comment:
Firstly, `getResponse` returns a Future. I could be wrong but looks like this example will not compile. 

Secondly, I don't think it is not the responsibility of this library to inform consumers on how to serialize Thrift based instances to JSON. They are free to choose any JSON serialization library.

Reply:
I think an example of how it _could_ be done is well worth including.

Reply:
I had intended to test this, but I'm getting more fundamental compilation errors while following the guidelines to trying the client out with `sbt console` - will investigate

Reply:
@anshulbajpai thanks for your comments!

Re: code, thanks for spotting. I'm just making the appropriate changes.

Re: library responsibility, this is an interesting point. My intent was to provide a convenient example for a common use-case, on the assumption that consumers will be interested in serialising our query responses and may not have strong opinions as to how they do it. (Or they do have strong opinions and are happy with Circe 😁 )

Because the models that CAPI returns are large and complex, my assumption is that using the decoders in the CAPI models library is the quickest way to achieve this goal. As the models library is a transitive dependency of this library, but users may not know to look there for this use-case, it felt appropriate to link to it here.

Consumers of this library may serialise to JSON in other ways, as you say – and perhaps we should be clear that this is possible – but should this should stop us from providing guidance for a common case?

Reply:
RE: _Or they do have strong opinions and are happy with Circe_ - You are right but most probably the consumers might already be using some other JSON serialization (e.g. if they are using Play framework)

@jonathonherbert - I think it is a good point that you mentioned that the CAPI models are large and complex and with such models, there is a heavy cost associated with JSON serialization/deserialization. 
And to get more understanding on this, do you expect to see any diff in the JSON output from CirceEncoder and the raw JSON body from HTTP response of the ContentAPI?

Reply:
Few interesting points from offline discussion with @jonathonherbert -

1. This library receives the CAPI response in Thrift format and not in JSON format (passed as query param to CAPI) which wasn't my understanding. This is why I initially found it strange that why we want to add serialization guidance when the CAPI returns JSON response.

2. @jonathonherbert requirement is JSON response and I could now see why he suggested adding the guidance for JSON serialization in this PR. This also means that we should add the support for returning raw JSON response so that consumers of this library do not have to deserialize to Thrift classes and then serialize those instances to JSON. I'll add a ticket for this in our backlog. Once we have the support for the raw JSON response, we wouldn't need this extra serialization guidance.

Reply:
Thanks for the useful discussion @anshulbajpai 👍